web/src/js/ducks/flows.js

```diff
@@ -20,32 +23,54 @@ const defaultState = {
     view: [],
     filter: undefined,
     highlight: undefined,
+    sort: {sortColumn: undefined, sortDesc: false},
 }
 
 function makeFilterFn(filter) {
     return filter ? Filt.parse(filter) : () => true;
 }
 
+
+function makeSortFn(sort){
+    let column = _.find(flowtable_columns, c => c.name == sort.sortColumn);
+    if (!column) return;
+
+    let sortKeyFun = column.sortKeyFun;
+    if (sort.sortDesc) {
+        sortKeyFun = sortKeyFun && function () {
+            const k = column.sortKeyFun.apply(this, arguments);
+            return _.isString(k) ? reverseString("" + k) : -k;
+        };
+    }
+    return sortKeyFun;
+}
+
 export default function reducer(state = defaultState, action) {
     switch (action.type) {
         case UPDATE_FLOWS:
             let all = reduceList(state.all, action)
             return {
                 ...state,
                 all,
-                view: updateViewList(state.view, state.all, all, action, makeFilterFn(action.filter))
+                view: updateViewList(state.view, state.all, all, action, makeFilterFn(action.filter), makeSortFn(state.sort))
             }
         case SET_FILTER:
             return {
                 ...state,
                 filter: action.filter,
-                view: updateViewFilter(state.all, makeFilterFn(action.filter))
+                view: updateViewFilterSort(state.all, makeFilterFn(action.filter), makeSortFn(state.sort))
             }
         case SET_HIGHLIGHT:
             return {
                 ...state,
                 highlight: action.highlight
             }
+        case SET_SORT:
+            return {
+                ...state,
+                sort: action.sort,
+                view: updateViewFilterSort(state.all, makeFilterFn(state.filter), makeSortFn(action.sort))
```

Comment:
I think we want two distinct functions `updateViewFilter` and `updateViewSort`, so that we don't have to `.filter()` again when we change the sort order. (see [previous comment](https://github.com/mitmproxy/mitmproxy/pull/1220/files#r65981944))